.github/workflows/main.yml

```
@@ -0,0 +1,86 @@
+name: CI
+
+on: pull_request
+
+jobs:
+  jjb:
+    runs-on: ubuntu-latest
+    strategy:
+      fail-fast: false
+    name: Jenkins Job Builder
+    steps:
+      - uses: actions/checkout@v2
+      - name: Setup python
+        uses: actions/setup-python@v1
+        with:
+          python-version: '2.7'
+      - name: Restore pip cache
+        uses: actions/cache@v2
+        with:
+          path: ~/.cache/pip
+          key: ${{ runner.os }}-pip-${{ hashFiles('**/requirements*.txt') }}
+          restore-keys: |
+            ${{ runner.os }}-pip-
+      - name: Install dependencies
+        run: |
+          pip install jenkins-job-builder
+          pip install lxml
+          pip install requests
+      - name: Run tests
+        run: |
+          for d in puppet/modules/jenkins_job_builder/files/theforeman.org; do
+            DIR="."
+            ( cd $d && jenkins-jobs -l debug test -r $DIR )
+            ( cd $d && jenkins-jobs test -r $DIR --config-xml -o output && find output -name config.xml -exec python ../jenkins-lint.py --xml {} + && rm -rf output )
+          done
+
+          for d in puppet/modules/jenkins_job_builder/files/centos.org; do
+            DIR="jobs"
+            ( cd $d && jenkins-jobs -l debug test -r $DIR )
+            ( cd $d && jenkins-jobs test -r $DIR --config-xml -o output && find output -name config.xml -exec python ../jenkins-lint.py --xml {} + && rm -rf output )
+          done
+  puppet:
+    runs-on: ubuntu-latest
+    strategy:
+      fail-fast: false
+    name: Puppet
+    steps:
+      - uses: actions/checkout@v2
+        with:
+          submodules: true
+      - name: Setup ruby
+        uses: actions/setup-ruby@v1
+        with:
+          ruby-version: '2.6'
+      - name: Install dependencies
+        run: |
+          sudo apt-get update && sudo apt-get install -y yajl-tools
+          gem install bundler
+          cd puppet
+          bundle config path ~/.vendor/bundle
+      - uses: actions/cache@v2
+        with:
+          path: ~/.vendor/bundle
+          key: ${{ runner.os }}-${{ matrix.ruby }}-gems-unit-
+          restore-keys: |
+            ${{ runner.os }}-${{ matrix.ruby }}-gems-unit-
+      - name: Bundle install
+        run: |
+          cd puppet
+          bundle install --jobs $(nproc) --retry 3
+      - name: Setup g10k
+        run: |
+          if [[ ! -f g10k ]] ; then
+            if [[ ! -f g10k-linux-amd64 ]] ; then
+              wget https://github.com/xorpaul/g10k/releases/download/v0.8.9/g10k-linux-amd64.zip
+            fi
+            unzip g10k-linux-amd64.zip
+          fi
```

We don't store this in git so it's safe to simplify it:
```suggestion
          wget https://github.com/xorpaul/g10k/releases/download/v0.8.9/g10k-linux-amd64.zip
          unzip g10k-linux-amd64.zip
```

For the Travis scripts I made it idempotent because I ran them locally as well.